sdk_extensions/resources/src/main/java/io/opentelemetry/sdk/extensions/resources/OsResource.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.extensions.resources;
+
+import io.opentelemetry.common.Attributes;
+import io.opentelemetry.sdk.resources.ResourceConstants;
+import io.opentelemetry.sdk.resources.ResourceProvider;
+import javax.annotation.Nullable;
+
+/** {@link ResourceProvider} which provides information about the current operating system. */
+public class OsResource extends ResourceProvider {
+
+  @Override
+  protected Attributes getAttributes() {
+    final String os;
+    try {
+      os = System.getProperty("os.name");
+    } catch (SecurityException t) {
+      // Security manager enabled, can't provide much os information.
+      return Attributes.empty();
+    }
+
+    if (os == null) {
+      return Attributes.empty();
+    }
+
+    Attributes.Builder attributes = Attributes.newBuilder();
+
+    String osName = getOs(os);
+    if (osName != null) {
+      attributes.setAttribute(ResourceConstants.OS_NAME, osName);
+    }
+
+    String version = null;
+    try {
+      version = System.getProperty("os.version");
+    } catch (SecurityException e) {
+      // Ignore
+    }
+    String osDescription = version != null ? os + ' ' + version : os;
+    attributes.setAttribute(ResourceConstants.OS_DESCRIPTION, osDescription);
+
+    return attributes.build();
+  }
+
+  @Nullable
+  private static String getOs(String os) {
```

Comment:
Would `contains` make the code more robust? I am not familiar with all the cases so I might be wrong.